### PR TITLE
4 create the base of the server

### PR DESCRIPTION
**Pull request by Peralban**
Create base of the server.
To obtains more details, please see:
- #4

### CODE REVIEW

File: Network/src/ClientList/client_list.c [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+        tmp->next->client = client;
+        tmp->next->next = NULL;
+        tmp->next->prev = tmp;
+    }
+    return 0;
+}
+
+static void remove_client_bis(client_list_t *tmp, client_t *client)
+{
+    while (tmp != NULL && tmp->client != client) {
```
Comment (aimericdrk) [inline Network/src/ClientList/client_list.c:54]: usless curly bracket

File: Network/Tests/src/ClientList/tests_linked_list.c [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+** test.c
+*/
+
+#include <criterion/criterion.h>
+#include <criterion/redirect.h>
+#include "ClientList/client_list.h"
+
+Test(create_client_list, test1)
+{
+    client_list_t *list = create_client_list();
```
Comment (aimericdrk) [inline Network/Tests/src/ClientList/tests_linked_list.c:14]: missing space after variable declaration

File: Network/Tests/src/ClientList/tests_linked_list.c [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+    if (list->client == NULL && list->next == NULL && list->prev == NULL)
+        cr_assert_eq(1, 1);
+    else
+        cr_assert_fail();
+}
+
+Test(create_client, test2)
+{
+    struct sockaddr_in *clientAddress = malloc(sizeof(struct sockaddr_in));
+    client_t *client = create_client(1, clientAddress);
```
Comment (aimericdrk) [inline Network/Tests/src/ClientList/tests_linked_list.c:24]: unverified malloc + missing blank line after variable declaration

File: Network/Tests/src/ClientList/tests_linked_list.c [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+        cr_assert_eq(1, 1);
+    else
+        cr_assert_fail();
+}
+
+Test(add_client_to_list, test3)
+{
+    client_list_t *list = create_client_list();
+    struct sockaddr_in *clientAddress = malloc(sizeof(struct sockaddr_in));
+    client_t *client = create_client(1, clientAddress);
```
Comment (aimericdrk) [inline Network/Tests/src/ClientList/tests_linked_list.c:35]: unverified malloc + missing blank line after variable declaration

File: Network/Tests/src/ClientList/tests_linked_list.c [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+        cr_assert_eq(1, 1);
+    else
+        cr_assert_fail();
+}
+
+Test(remove_client_from_list, test4)
+{
+    client_list_t *list = create_client_list();
+    struct sockaddr_in *clientAddress = malloc(sizeof(struct sockaddr_in));
+    client_t *client = create_client(1, clientAddress);
```
Comment (aimericdrk) [inline Network/Tests/src/ClientList/tests_linked_list.c:47]: same

File: Network/src/ClientList/client_list.c [hunk 6]
```diff
@@ -0,0 +1,79 @@
...
+
+    if ((*list)->client == NULL) {
+        (*list)->client = client;
+        (*list)->next = NULL;
+        (*list)->prev = NULL;
+    } else {
+        while (tmp->next != NULL)
+            tmp = tmp->next;
+        tmp->next = malloc(sizeof(client_list_t));
+        if (tmp->next == NULL) {
```
Comment (aimericdrk) [inline Network/src/ClientList/client_list.c:42]: usless curly bracket

**Review by Neiluge — APPROVED** [review]

File: Network/src/Server/server_loop.c [hunk 7]
```diff
@@ -0,0 +1,84 @@
...
+    buffer[buffer_length] = '\0';
+    printf("Received: %s\n", buffer);
+    send(tmp->client->clientServer->socket, buffer, buffer_length, 0);
+}// the send is temporary, it will be deplaced in another function,
+
+// server is not used for no, but it will be used in the future
+static void client_already_connected(client_list_t **list, server_t **server)
+{
+    for (client_list_t *tmp = *list; tmp->client != NULL; tmp = tmp->next) {
+        if (FD_ISSET(tmp->client->clientServer->socket, &(*server)->readfds)) {
```
Comment (aimericdrk) [inline Network/src/Server/server_loop.c:42]: usless curly bracket

**Review by aimericdrk — APPROVED** [review]

**Review by Antoinegtir — APPROVED** [review]

**Review by landryarki — APPROVED** [review]
:crown:

**Pull request merged**